Documentation/scalability/scalability.rst

```diff
@@ -0,0 +1,397 @@
+now have **250 different network policies selecting 10000 pods and 250 different
+network policies selecting 50000 pods on a cluster with 1000 nodes.** Similarly
+to the previous step we will deploy 125 L4 policies and another 125 L7 policies.
+
+.. figure:: images/image_9_01.png
+.. figure:: images/image_9_02.png
+
+Similar to the creation of the previous 250 CNPs, there was also an increase in
+CPU usage during the creation of the CCNPs. The CPU usage was similar even
+though the policies were effectively selecting more pods.
+
+.. figure:: images/image_9_03.png
+
+As all pods running in a node are selected by **all the 250 CCNPs created**, we
+see an increase of the **Endpoint regeneration time** which **peaked a little
+above 3s.**
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+10. "Accidentally" delete 10000 pods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this step we have "accidentally" deleted 10000 random pods. Kubernetes will
+then recreate 10000 new pods so it will help us understand what the convergence
+time is for all the network polices deployed.
+
+.. figure:: images/image_10_01.png
+.. figure:: images/image_10_02.png
+
+
+* The first mark represents the point in time when pods were "deleted" and the
+  second mark represents the point in time Kubernetes finished recreating 10k
+  pods.
+
+* Besides the CPU usage slightly increasing while pods are being scheduled in
+  the cluster we did see some interesting data points in the BPF memory usage.
+  As each endpoint can have one or more dedicated BPF maps, the BPF memory usage
+  is directly proportional to the number of pods running in a node. **If the
+  number of pods per node decreases so does the BPF memory usage.**
+
+.. figure:: images/image_10_03.png
+
+We inferred the time it took for all the endpoints to get regenerated by looking
+at the number of Cilium endpoints with the policy enforced over time.
+Luckily enough we had another metric that was showing how many Cilium endpoints
+had policy being enforced:
+
+.. figure:: images/image_10_04.png
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+11. Control plane metrics over the test run
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The focus of this test was to study the Cilium agent resource consumption at
+scale. However we also monitored some metrics of the control plane nodes such as
+etcd metrics and CPU usage of the k8s-controllers and we present them in the
+next figures.
+
+.. figure:: images/image_11_01.png
+
+Memory consumption of the 3 etcd instances during the entire scalability
+testing.
+
+.. figure:: images/image_11_02.png
+
+CPU Usage for the 3 controller nodes, average latency per request type in
+the etcd cluster as well as the number of operations per second made to etcd.
+
+.. figure:: images/image_11_03.png
+
+All etcd metrics, from left to right, from top to bottom: database size,
+disk sync duration, client traffic in, client traffic out, peer traffic in,
+peer traffic out.
```

Comment:
I don't think that needs to be stated.

Reply:
It just to not left this image without a description.